tests/testImageMapReduce.py

```diff
@@ -0,0 +1,243 @@
+    """
+    mapperSubtask = pexConfig.ConfigurableField(
+        doc="Mapper subtask to run on each subimage",
+        target=TestImageMapperSubtask,
+    )
+
+
+class TestImageMapperSubtask2(ImageMapperSubtask):
+    ConfigClass = TestImageMapperSubtaskConfig
+    _DefaultName = "ip_diffim_TestImageMapperSubtask2"
+
+    def __init__(self, *args, **kwargs):
+        """! Create the image gridding subTask
+        @param *args arguments to be passed to lsst.pipe.base.task.Task.__init__
+        @param **kwargs keyword arguments to be passed to lsst.pipe.base.task.Task.__init__
+        """
+        ImageMapperSubtask.__init__(self, *args, **kwargs)
+
+    def run(self, subExp, expandedSubExp, fullBBox, **kwargs):
+        """!Return the mean of the given subExposure.
+
+        @param[in] subExp the sub-exposure of `exposure` upon which to operate
+        @param[in] expandedSubExp the expanded sub-exposure of `exposure` upon which to operate
+        @param[in] fullBBox the bounding box of the original exposure
+        @return a float, the mean of `subExp.getMaskedImage().getArray()`.
+        """
+        return subExp.getMaskedImage().getImage().getArray().mean()
+
+
+class TestImageMapReduceConfig2(ImageMapReduceConfig):
+    """!
+    \anchor TestImageMapReduceConfig_
+
+    \brief Configuration parameters for the TestImageMapReduceTask2
+    """
+    mapperSubtask = pexConfig.ConfigurableField(
+        doc="Mapper subtask to run on each subimage",
+        target=TestImageMapperSubtask2,
+    )
+
+
+class ImageMapReduceTest(lsst.utils.tests.TestCase):
+    """!A test case for the image gridded processing task
+    """
+
+    def setUp(self):
+        self.exposure = afwImage.ExposureF(128, 128)
+        self.exposure.setPsf(measAlg.DoubleGaussianPsf(11, 11, 2.0, 3.7))
+        mi = self.exposure.getMaskedImage()
+        mi[:, :] = 0.
+
+    def tearDown(self):
+        del self.exposure
+
+    def testExampleTaskNoOverlaps(self):
+        """Test sample grid task that adds 5.0 to input image and uses
+        default 'copy' `reduceOperation`.
+        """
+        config = TestImageMapReduceConfig()
+        task = ImageMapReduceTask(config)
+        config.mapperSubtask.addAmount = 5.
+        newExp = task.run(self.exposure)
+        newMI = newExp.getMaskedImage()
+        newArr = newMI.getImage().getArray()
+        mi = self.exposure.getMaskedImage()
+
+        self.assertClose(mi.getImage().getArray().mean(), newArr.mean() - 5.)
+        self.assertClose(newArr.mean(), 5.)
+        self.assertClose(newArr.min(), 5.)
+        self.assertClose(newArr.max(), 5.)
+
+    def testExampleTaskWithOverlaps(self):
+        """Test sample grid task that adds 5.0 to input image and uses
+        'average' `reduceOperation`.
+        """
+        config = TestImageMapReduceConfig()
+        config.gridStepX = config.gridStepY = 8
+        config.reducerSubtask.reduceOperation = 'average'
```

Comment:
Yes. Difference is with 'average' and when the grid elements overlap, it will take the average of the returned values for areas that contain overlapping regions. Thus for this simple case it produces the same exact result as the no overlap test.